devtools/projects/ng-devtools/src/lib/devtools-tabs/devtools-tabs.component.ts

```diff
@@ -63,11 +73,14 @@ export class DevToolsTabsComponent {
 
   readonly snapToRoot = signal(false);
 
-  readonly tabs = computed<Tabs[]>(() => {
-    const alwaysShown: Tabs[] = ['Components', 'Profiler', 'Injector Tree'];
-    return this.routerGraphEnabled() && this.routes().length > 0
-      ? [...alwaysShown, 'Router Tree']
-      : alwaysShown;
+  readonly tabs = computed<Tab[]>(() => {
+    const supportedApis = this.supportedApis();
+    const apiBound: ApiBoundTab[] = API_BOUND_TABS.filter((tab) => supportedApis[TAB_TO_API[tab]]);
+    const tabs: Tab[] = ['Components', ...apiBound];
+
+    return supportedApis.routes && this.routerGraphEnabled() && this.routes().length > 0
+      ? [...tabs, 'Router Tree']
+      : tabs;
   });
```

Comment:
Consider: Is this more generic than it needs to be? Would things be simpler just hard-coding for each tab?

```suggestion
  readonly tabs = computed<Tab[]>(() => {
    const supportedApis = this.supportedApis();
    
    const tabs = ['Components'];
    if (supportedApis.profiler) tabs.push('Profiler');
    if (supportedApis.dependencyInjection) tabs.push('Injector Tree');
    if (supportedApis.routes && this.routerGraphEnabled() && this.routes().length > 0) tabs.push('Router Tree');
    
    return tabs;
  });
```

My thinking here is:
1. We're not likely to add so many tabs that this becomes infeasible to manage.
2. `routes` needs an extra condition anyways, so the current generic approach isn't actually that generic.
3. No more need to separate `Components` from `API_BOUND_TABS`, they're all just "tabs".
4. Removes `supportedApis[TAB_TO_API[tab]]` which is pretty tricky to follow IMHO.

Reply:
:+1: My initial version of the change included the Components as part of the supported APIs, so only the Routes was the odd one with the additional check. But I agree that with the current amount of tabs and the high unlikeliness for them to grow in number makes this approach pointless and harder to follow.